packages/eslint-plugin-pf-codemods/src/rules/v6/chipReplaceWithLabel/chip-replace-with-label.ts

```diff
@@ -0,0 +1,202 @@
+  return [
+    fixer.replaceText(importSpecifier, newName),
+    fixer.replaceText(node.source, "'@patternfly/react-core'"),
+  ];
+};
+
+const getJSXAttribute = (node: JSXElement, attributeName: string) => {
+  return node.openingElement.attributes.find(
+    (attribute) =>
+      attribute.type === 'JSXAttribute' && attribute.name.name === attributeName
+  ) as JSXAttribute | undefined;
+};
+
+const parseBadgeAttributeValue = (
+  badgeAttributeValue: JSXAttributeValue
+): Node => {
+  if (badgeAttributeValue.type !== 'JSXExpressionContainer') {
+    return badgeAttributeValue;
+  }
+
+  const isValueJSX = ['JSXElement', 'JSXEmptyExpression'].includes(
+    badgeAttributeValue.expression.type
+  );
+
+  return isValueJSX ? badgeAttributeValue.expression : badgeAttributeValue;
+};
+
+const moveBadgeAttributeToBody = (
+  node: JSXElement,
+  fixer: Rule.RuleFixer,
+  context: Rule.RuleContext
+) => {
+  const badgeAttribute = getJSXAttribute(node, 'badge');
+
+  return badgeAttribute
+    ? [
+        fixer.insertTextBefore(
+          node.closingElement as JSXClosingElement,
+          context
+            .getSourceCode()
+            .getText(
+              badgeAttribute?.value
+                ? parseBadgeAttributeValue(badgeAttribute.value)
+                : undefined
+            )
+        ),
+        fixer.remove(badgeAttribute),
+      ]
+    : [];
+};
+
+const renameOnClickAttribute = (node: JSXElement, fixer: Rule.RuleFixer) => {
+  const onClickAttribute = getJSXAttribute(node, 'onClick');
+
+  return onClickAttribute
+    ? [fixer.replaceText(onClickAttribute.name, 'onClose')]
+    : [];
+};
+
+// https://github.com/patternfly/patternfly-react/pull/10049
+module.exports = {
+  meta: { fixable: 'code' },
+  create: function (context: Rule.RuleContext) {
+    const { imports } = getFromPackage(
+      context,
+      '@patternfly/react-core/deprecated'
+    );
+
+    const chipImport = imports.find(
+      (specifier) => specifier.imported.name === 'Chip'
+    );
+
+    const chipGroupImport = imports.find(
+      (specifier) => specifier.imported.name === 'ChipGroup'
+    );
+
+    return !chipImport && !chipGroupImport
+      ? {}
+      : {
+          ImportDeclaration(node: ImportDeclaration) {
+            context.report({
+              node,
+              message: `Chip has been deprecated. Running the fix flag will replace Chip and ChipGroup components with Label and LabelGroup components respectively.`,
+              fix(fixer) {
+                return [
+                  ...renameImportSpecifier(chipImport, node, fixer, 'Label'),
+                  ...renameImportSpecifier(
+                    chipGroupImport,
+                    node,
+                    fixer,
+                    'LabelGroup'
+                  ),
+                ];
+              },
+            });
+          },
+
+          JSXElement(node: JSXElement) {
+            if (node.openingElement.name.type === 'JSXIdentifier') {
```

Comment:
Nit: I'd refactor this to a guard clause since all the rest of the logic is encapsulated in it.